src/Service/Services/RequestValidator.cs

```diff
@@ -217,56 +219,57 @@ public static JsonElement ParseRequestBody(string requestBody)
-                    statusCode: HttpStatusCode.BadRequest,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
-            }
+                case Config.Operation.Insert:
+                    if (!isPrimaryKeyRouteEmpty)
+                    {
+                        throw new DataApiBuilderException(
+                            message: PRIMARY_KEY_INVALID_USAGE_ERR_MESSAGE,
+                            statusCode: HttpStatusCode.BadRequest,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                    }
 
-            return ParseRequestBody(requestBody);
-        }
+                    if (!isQueryStringEmpty)
+                    {
+                        throw new DataApiBuilderException(
+                            message: QUERY_STRING_INVALID_USAGE_ERR_MESSAGE,
+                            statusCode: HttpStatusCode.BadRequest,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                    }
 
-        /// <summary>
-        /// Validates the primarykeyroute is populated with respect to a Delete operation.
-        /// </summary>
-        /// <param name="primaryKeyRoute">URL route e.g. "Entity/id/1"</param>
-        /// <exception cref="DataApiBuilderException">Raised when primary key route is absent</exception>
-        public static void ValidateDeleteRequest(string? primaryKeyRoute)
-        {
-            if (string.IsNullOrWhiteSpace(primaryKeyRoute))
-            {
-                throw new DataApiBuilderException(
-                    message: PRIMARY_KEY_NOT_PROVIDED_ERR_MESSAGE,
-                    statusCode: HttpStatusCode.BadRequest,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                    break;
+                case Config.Operation.Delete:
+                case Config.Operation.Update:
+                case Config.Operation.UpdateIncremental:
+                case Config.Operation.Upsert:
+                case Config.Operation.UpsertIncremental:
+                    /// Validate that the primarykeyroute is populated for these operations.
+                    if (isPrimaryKeyRouteEmpty)
+                    {
+                        throw new DataApiBuilderException(
+                            message: PRIMARY_KEY_NOT_PROVIDED_ERR_MESSAGE,
+                            statusCode: HttpStatusCode.BadRequest,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                    }
+
+                    break;
+                default:
+                    throw new DataApiBuilderException(
+                        message: "Unexpected operation encountered. Cannot perform validations on URL components.",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
```

Comment:
Don't we validate the operationType in another location? If you need to keep this, I would avoid intentionally throwing a HTTP 5XX error. This error could be corrected by the user, right? I would suggest using a 4XX (probably 400 bad request) here.

Reply:
restservice line 160:
```csharp
                        throw new DataApiBuilderException(
                            message: "This operation is not supported.",
                            statusCode: HttpStatusCode.BadRequest,
                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
                }
```

Reply:
Not really, with the current code we will not hit the default case at all. If someone later uses this function for a new operation, not present here in the switch case, then we would throw this internal server error. For users of DAB, this case should never be hit.